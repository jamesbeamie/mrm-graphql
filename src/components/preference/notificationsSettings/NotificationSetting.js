import React from 'react';
import Switch from 'react-toolbox/lib/switch';
import PropTypes from 'prop-types';
import '../../../assets/styles/notificationSettingsList.scss';

/**
 * will render a single notification setting card
 *
 * @param {Object} notificationSettingsObject
 *
 * @returns {JSX}
 *
 */
const NotificationSetting = ({
  title, body, toggle, handleChange,
}) => (
  <div className="notification-container">
    <section className="notification-content">
      <h2 className="notification-title">{title}</h2>
      <p>{body}</p>
    </section>
    <section className="notification-switch">
      <Switch
        id="switchBtn"
        className="switch-component"
        checked={toggle}
        onChange={handleChange}
      />
    </section>
  </div>
);

NotificationSetting.propTypes = {
  title: PropTypes.string.isRequired,
  body: PropTypes.string.isRequired,
  toggle: PropTypes.bool.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default NotificationSetting;
