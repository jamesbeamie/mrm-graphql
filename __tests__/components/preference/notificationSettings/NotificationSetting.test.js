import React from 'react';
import { shallow } from 'enzyme';
import NotificationSetting from '../../../../src/components/preference/notificationsSettings/NotificationSetting';

describe('NotificationSetting Component', () => {
  it('renders correctly from memory', () => {
    expect(shallow(<NotificationSetting
      title="Meeting Notifications"
      body="some"
      toggle={false}
      handleChange={() => ''}
    />)).toMatchSnapshot();
  });
});
