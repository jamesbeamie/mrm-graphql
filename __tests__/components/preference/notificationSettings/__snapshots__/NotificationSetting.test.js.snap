// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSetting Component renders correctly from memory 1`] = `
<div
  className="notification-container"
>
  <section
    className="notification-content"
  >
    <h2
      className="notification-title"
    >
      Meeting Notifications
    </h2>
    <p>
      some
    </p>
  </section>
  <section
    className="notification-switch"
  >
    <ThemedSwitch
      checked={false}
      className="switch-component"
      composeTheme="deeply"
      disabled={false}
      id="switchBtn"
      mapThemrProps={[Function]}
      onChange={[Function]}
    />
  </section>
</div>
`;
